src/ui/components/shared/PortalDropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { CSSProperties, PureComponent } from "react";
+import { createPortal } from "react-dom";
+import { DropdownProps } from "./Dropdown";
+import "./PortalDropdown.css";
+
+interface PortalDropdownProps extends DropdownProps {
+  distance?: number;
+}
+
+export default class PortalDropdown extends PureComponent<PortalDropdownProps> {
```

Comment:
minor thing here, but this looks like an opportunity for a functional component. We've been using a lot of them in the `src/ui` directory and it has felt cleaner.

Reply:
But where can I store the `buttonRef`? And how do I replace the callbacks without accidentally triggering lots of renders?